### PR TITLE
[PHPCS] Enable Sider's recommended set of rules

### PR DESCRIPTION
**Pull request by sleeeto**
> Explain a summary, purpose, or background for this change.

Enable Sider's recommended ruleset of 'code sniffer'

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

None

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

**Comment by sleeeto** [comment]
@ybiquitous @yykamei Please check it.

File: lib/runners/processor/code_sniffer.rb [hunk 1]
```diff
@@ -29,10 +29,12 @@ class Processor::CodeSniffer < Processor
 
     register_config_schema(name: :code_sniffer, schema: Schema.runner_config)
 
+    DEFAULT_CONFIG_FILE = (Pathname(Dir.home) / "sider_recommended_code_sniffer.xml").to_path.freeze
+
     DefaultOptions = _ = Struct.new(:standard, :extensions, :target, keyword_init: true)
     OPTIONS_CAKE_PHP = DefaultOptions.new(standard: "CakePHP", extensions: "php", target: "app/").freeze
     OPTIONS_SYMFONY = DefaultOptions.new(standard: "Symfony", extensions: "php", target: "src/").freeze
-    OPTIONS_PSR2 = DefaultOptions.new(standard: "PSR2", extensions: "php", target: "./").freeze
+    OPTIONS_PSR2 = DefaultOptions.new(standard: DEFAULT_CONFIG_FILE, extensions: "php", target: "./").freeze
```
Comment (yykamei) [inline lib/runners/processor/code_sniffer.rb:37]: Is this OK to remain the constant name as `OPTIONS_PSR2` even though the `standard` is not `"PSR2"`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ Misc:
 - **ESLint** Update pre-installed packages [#2155](https://github.com/sider/runners/pull/2155)
 - **Flake8** Enable new recommended configuration [#2157](https://github.com/sider/runners/pull/2157)
 - Simplify generated `Gemfile` [#2168](https://github.com/sider/runners/pull/2168)
+- **PHP_CodeSniffer** Enable new recommended configuration [#2157](https://github.com/sider/runners/pull/2182)
```
Comment (ybiquitous) [inline CHANGELOG.md:36]:
```suggestion
- **PHP_CodeSniffer** Enable new recommended configuration [#2182](https://github.com/sider/runners/pull/2182)
```

File: lib/runners/processor/code_sniffer.rb (hunk 1 above)
Comment (ybiquitous) [reply lib/runners/processor/code_sniffer.rb:37]: @sleeeto It's confusing. Please rename the constant.

```suggestion
    OPTIONS_DEFAULT = DefaultOptions.new(standard: DEFAULT_CONFIG_FILE, extensions: "php", target: "./").freeze
```

Comment (sleeeto) [reply lib/runners/processor/code_sniffer.rb]: OK, I renamed it.

File: CHANGELOG.md (hunk 2 above)
Comment (sleeeto) [reply CHANGELOG.md]: I fixed it.

**Review by ybiquitous — APPROVED** [review]
Thanks. LGTM 👍

**Review by yykamei — APPROVED** [review]
LGTM

**Pull request merged**